### PR TITLE
FOGL-7948: Show expand/collapsed icon only when node has children on configuration page

### PR DESCRIPTION
**Pull request by monika-Dianomic**

### CODE REVIEW

**Review by mshariq-nerd — CHANGES_REQUESTED** [review]
SOUTH, NORTH  & NOTIFICATIONS need to be filtered from the configuration tree.  

That was the code for filtering SOUTH, NORTH  & NOTIFICATIONS.
```
this.rootCategories = this.rootCategories.filter(el => el.key.toUpperCase() !== 'SOUTH')
              .filter(el => el.key.toUpperCase() !== 'NORTH')
              .filter(el => el.key.toUpperCase() !== 'NOTIFICATIONS');
```

File: src/app/components/core/configuration-manager/configuration-manager.component.ts [hunk 1]
```diff
@@ -40,92 +38,57 @@ export class ConfigurationManagerComponent implements OnInit {
...
-                id: element.key,
-                name: element.description,
-                description: element.description,
-                hasChildren: true, children: []
-              });
+            _nodes.push({
+              id: element.key,
+              name: (element.hasOwnProperty('displayName')) ? element.displayName : element.description,
+              description: element.description,
+              hasChildren: (element.children.length) > 0 ? true : false , children: this.addNameKey(element.children)
```
Comment (praveen-garg) [inline src/app/components/core/configuration-manager/configuration-manager.component.ts:68]: move children key value to next line after

`hasChildren: element.children.length > 0,`

File: src/app/components/core/configuration-manager/configuration-manager.component.ts [hunk 2]
```diff
@@ -40,92 +38,57 @@ export class ConfigurationManagerComponent implements OnInit {
...
+            });
+          });
+
+          // If category 'General' exists, show it on index 0
```
Comment (praveen-garg) [inline src/app/components/core/configuration-manager/configuration-manager.component.ts:72]: Category group

File: src/app/components/core/configuration-manager/configuration-manager.component.ts [hunk 3]
```diff
@@ -139,56 +102,22 @@ export class ConfigurationManagerComponent implements OnInit {
...
-            });
-          }, error => {
-            if (error.status === 0) {
-              console.log('service down ', error);
-            } else {
-              this.alertService.error(error.statusText);
-            }
-          });
-    }
+  addNameKey(childrenData) {
```
Comment (praveen-garg) [inline src/app/components/core/configuration-manager/configuration-manager.component.ts:105]: What is addNameKey?

File: src/app/components/core/configuration-manager/configuration-manager.component.ts [hunk 4]
```diff
@@ -15,9 +15,7 @@ import {
 
 export class ConfigurationManagerComponent implements OnInit {
   public categoryData = [];
-  public rootCategories = [];
   public JSON;
```
Comment (praveen-garg) [inline src/app/components/core/configuration-manager/configuration-manager.component.ts:18]: What is this variable for?

File: src/app/components/core/configuration-manager/configuration-manager.component.ts (hunk 3 above)
Comment (monika-Dianomic) [reply src/app/components/core/configuration-manager/configuration-manager.component.ts:105]: `tree-root` component works on the basis of `name` key. And we don't have `name` key in the object received from API. So I've added it.

File: src/app/components/core/configuration-manager/configuration-manager.component.ts (hunk 4 above)
Comment (monika-Dianomic) [reply src/app/components/core/configuration-manager/configuration-manager.component.ts:18]: It seems, it doesn't have any use here but need to verify once more. It is as it was before.

File: src/app/components/core/configuration-manager/configuration-manager.component.ts [hunk 5]
```diff
@@ -34,159 +31,86 @@ export class ConfigurationManagerComponent implements OnInit {
...
+    if (['Advanced', 'General', 'Utilities'].includes(rootId)) {
+      if (tree?.treeModel) {
+        const nodes = tree.treeModel.nodes;
+        const node = nodes.find(c => c.id == rootId)?.children[0];
+        const firstChild = tree.treeModel.getNodeById(node.id);
+        firstChild.setActiveAndVisible();
+        this.selectedNode = {id: node.id, description: node.description};
+        this.getCategory(node.id, node.description);
+      }
+    } else {
```
Comment (praveen-garg) [inline src/app/components/core/configuration-manager/configuration-manager.component.ts:109]: why the if (tree?.treeModel) { check is not required here, if in the if block above?

File: src/app/components/core/configuration-manager/configuration-manager.component.ts [hunk 6]
```diff
@@ -257,14 +182,15 @@ export class ConfigurationManagerComponent implements OnInit {
         this.validConfigForm = false;
         this.alertService.success('Configuration updated successfully.', true);
         this.ngProgress.done();
+        this.selectedNode = {id: categoryName, description: categoryDescription};
```
Comment (praveen-garg) [inline src/app/components/core/configuration-manager/configuration-manager.component.ts:185]: what if  categoryDescription is missing?

**Review by praveen-garg — COMMENTED** [review]
sorting please.

File: src/app/components/core/configuration-manager/configuration-manager.component.ts (hunk 6 above)
Comment (monika-Dianomic) [reply src/app/components/core/configuration-manager/configuration-manager.component.ts:185]: `categoryDescription` will always present there.

**Review by mshariq-nerd — APPROVED** [review]

**Pull request merged**